File: Advent of Code 2022/Day 1/1b.py
import heapq


def main():
    input: list[str] = open("Advent of Code 2022\Day 1\input.txt").read().split("\n\n")

    items: list[int] = [0, 0, 0]
    for item in input:
        val: int = 0
        for value in item.split("\n"):
            val += int(value)

        heapq.heappushpop(items, val)

    print(sum(items))
    return sum(items)


if __name__ == "__main__":
    main()

File: Advent of Code 2022/Day 1/1b.js
const fs = require("fs");

const input = fs.readFileSync("Advent of Code 2022/Day 1/input.txt", "utf-8");
const arr = input.split("\r\n\r\n");

let max_cal = [0, 0, 0];
let total = 0;
for (let item = 0; item < arr.length; item++) {
	const element = arr[item].split("\r\n");

	let sum = 0;
	for (let cal = 0; cal < element.length; cal++) {
		let val = parseInt(element[cal]);
		sum += val;
	}

	for (let cal = 0; cal < max_cal.length; cal++) {
		let val = max_cal[cal];
		if (sum > val) {
			max_cal[cal] = sum;
			max_cal.sort();
			break;
		}
	}
}

max_cal.forEach((num) => (total += num));
console.log(total);
